src/transform/groovy/ga/ozli/minecraftmods/groovylicious/transform/registroid/RegistroidASTTransformer.groovy

```diff
@@ -0,0 +1,391 @@
+                return true
+            }
+            types.add(drType)
+        }
+        return false
+    }
+
+    private FieldNode generateDR(final ClassNode targetClass, final FieldNode registryField, final String modId, boolean addToClass = true) {
+        if (registryField.type == RESOURCE_KEY_TYPE) {
+            makeResourceKey(targetClass, registryField, modId, addToClass)
+        } else if (registryField.type.isDerivedFrom(REGISTRY_TYPE)) {
+            makeRegistry(targetClass, registryField, modId, addToClass)
+        } else if (GeneralUtils.isOrImplements(registryField.type, FORGE_REGISTRY_TYPE)) {
+            makeForgeRegistry(targetClass, registryField, modId, addToClass)
+        } else {
+            addError("No known way of representing registry from ${registryField.owner.name}.${registryField.name}", registryField)
+            return null
+        }
+    }
+
+    private static FieldNode makeResourceKey(final ClassNode targetClass, final FieldNode registryField, final String modId, boolean addToClass = true) {
+        // ResourceKey<Registry<T>>
+        final type = registryField.type.genericsTypes[0].type.genericsTypes[0].type
+        final field = new FieldNode(
+                registryField.name.replace('_REGISTRY', 'S').toUpperCase(Locale.ROOT),
+                Opcodes.ACC_STATIC | Opcodes.ACC_PUBLIC | Opcodes.ACC_FINAL,
+                GenericsUtils.makeClassSafeWithGenerics(DeferredRegister, type), targetClass,
+                GeneralUtils.callX(
+                        DEFERRED_REGISTER_TYPE, 'create', GeneralUtils.args(
+                        GeneralUtils.fieldX(registryField), GeneralUtils.constX(modId)
+                    )
+                )
+        )
+        field.addAnnotation(TransformUtils.GENERATED_ANNOTATION)
+        if (addToClass) {
+            targetClass.addField(field)
+        }
+        return field
+    }
+
+    private static FieldNode makeForgeRegistry(final ClassNode targetClass, final FieldNode registryField, final String modId, boolean addToClass = true) {
+        // ForgeRegistry<T>
+        final type = registryField.type.genericsTypes[0].type
+        final field = new FieldNode(
+                registryField.name.toUpperCase(Locale.ROOT),
+                Opcodes.ACC_STATIC | Opcodes.ACC_PUBLIC | Opcodes.ACC_FINAL,
+                GenericsUtils.makeClassSafeWithGenerics(DeferredRegister, type),
+                targetClass, GeneralUtils.callX(
+                DEFERRED_REGISTER_TYPE, 'create', GeneralUtils.args(
+                    GeneralUtils.fieldX(registryField), GeneralUtils.constX(modId)
+                )
+            )
+        )
+        field.addAnnotation(TransformUtils.GENERATED_ANNOTATION)
+        if (addToClass) {
+            targetClass.addField(field)
+        }
+        return field
+    }
+
+    private static FieldNode makeRegistry(final ClassNode targetClass, final FieldNode registryField, final String modId, boolean addToClass = true) {
```

Comment:
While that is true, it would lead to multiple duplicate checks, and frankly, the amount of duplicated code is little, to which point it is not worth it.